plugins/processors/t128_filter/t128_filter.go

```diff
@@ -97,24 +157,82 @@ func (c orConjMatcher) Matches(point telegraf.Metric) bool {
 func createMatcher(conditions []Condition) (matcher, error) {
 	conditionMatchers := make([]matcher, len(conditions))
 	for i, condition := range conditions {
-		tagMatchers := getTagMatchers(condition.Tags)
+		tagMatchers, err := getTagMatchers(condition.Tags, condition.Mode)
+		if err != nil {
+			return nil, err
+		}
 
 		conditionMatchers[i] = andConjMatcher{matchers: tagMatchers}
 	}
 
 	return orConjMatcher{conditionMatchers}, nil
 }
 
-func getTagMatchers(tags tags) []matcher {
+func getTagMatchers(tags tags, mode mode) ([]matcher, error) {
 	tagMatchers := make([]matcher, len(tags))
 
 	j := 0
```

Comment:
I'm just noticing this now. I think avoiding this index would be less error prone.
```go
tagMatchers := make([]matcher, 0, len(tags))
...
tagMatchers = append(tagMatchers, matcher)
```